lib/exception_notifier.rb

```diff
@@ -96,8 +104,17 @@ def notifiers
     #   ExceptionNotifier.ignore_if do |exception, options|
     #     not Rails.env.production?
     #   end
-    def ignore_if(&block)
-      @@ignores << block
+    #
+    # It also accepts a condition that is specific to each individual notifier.
+    #   ExceptionNotifier.ignore_if(notifier: :email) do |exception, options|
+    #     not Rails.env.production?
+    #   end
+    def ignore_if(notifier: nil, &block)
```

Comment:
What do you think about having a separate method for ignoring notifiers? (vs adding it to the existing `ignore_if`),
I think having something like `ignore_notifier_if` would improve readability a bit, it'd also be self explanatory so we could do away with the new comment.

Reply:
@nicolasferraro,

Thanks for your quick feedback!
That't a fair point. I actually wondered a bit about that.

I thought we might (ultimately) want to DRY up two variables -  `@@ignores` and `@@by_notifier_ignores` altogether, as the first variable stores what should be ignored _globally_ while the latter is to be ignored _by a particular notifier_. By doing so, we could maybe give one unified api for the both `ignore_if` and `ignore_notifier_if` config settings, since we already have four different `ignore` options that feels a bit too much.

But yes, that maybe was overthinking, or at best, too early to discuss;) The two ignore's essentially have different effects in different phases, so that's reasonable to treat them separately for better clarity.

Let me refactor is as per your input!